### ISSUE TITLE
Bug in nested loop with await/defer.

### ISSUE THREAD

**Issue by jfgorski** [maxtaco/coffee-script#161]
I came across this bug with nested loops which have different range directions. Seems like the problem is that _positive is given scope outside all the loops, but each loop updates it.

```
for i in [0..1]
  for j in [1..0]
    console.log i,j
    await setTimeout defer(), 1
```

gives

0 1
0 0

but take out the await and get

0 1
0 0
1 1
1 0

**Comment by maxtaco** [comment]
Which version?

**Comment by jfgorski** [comment]
Originally with iced-coffee-script@1.8.0-e but also with latest on npm iced-coffee-script@108.0.8.

**Comment by maxtaco** [comment]
Thanks for the bug report, I'll take a look.

**Comment by maxtaco** [comment]
I have a fix in 0a28c88, but i should write some tests before i cut a release.  Thanks!

**Comment by jfgorski** [comment]
Cool. Thanks for iced btw. It's made CS so much more pleasant to use!

**Comment by maxtaco** [comment]
You're most welcome!

**Comment by maxtaco** [comment]
Closed in v108.0.9
